core/runtime/binaryen/runtime_api/core_impl.cpp

```diff
@@ -15,10 +15,11 @@ namespace kagome::runtime::binaryen {
   using primitives::Version;
 
   CoreImpl::CoreImpl(
-      const std::shared_ptr<RuntimeManager> &runtime_manager,
+      std::shared_ptr<WasmProvider> wasm_provider,
+      std::shared_ptr<RuntimeManager> runtime_manager,
       std::shared_ptr<storage::changes_trie::ChangesTracker> changes_tracker,
       std::shared_ptr<blockchain::BlockHeaderRepository> header_repo)
-      : RuntimeApi(runtime_manager),
+      : RuntimeApi(std::move(wasm_provider), std::move(runtime_manager)),
```

Comment:
We don't make sptr-member here. I think better get and use sprt by const-ref.

Reply:
Sorry, could you elaborate on that? There's a shared ptr member in Runtime API for this, it's moved there in the end.

Reply:
Here we are creating temporary object and moving it into RuntimeApi. If RuntimeApi receive it by reference - we doing redundant work. If we would use reference, we avoid temporary object making, and RuntimeApi will make it if needs.